govcd/system.go

```diff
@@ -975,3 +975,29 @@ func (vcdCli *VCDClient) GetOrgList() (*types.OrgList, error) {
 	}
 	return orgList, nil
 }
+
+// QueryAdminOrgVdcStorageProfileByID finds a StorageProfile of VDC by ID as admin
+func QueryAdminOrgVdcStorageProfileByID(vcdCli *VCDClient, id string) ([]*types.QueryResultAdminOrgVdcStorageProfileRecordType, error) {
```

Comment:
Thanks for this. Now just pardon, but I just tried to play with code and one thing hit me.
I think the signature could be:
```suggestion
func QueryAdminOrgVdcStorageProfileByID(vcdCli *VCDClient, id string) (*types.QueryResultAdminOrgVdcStorageProfileRecordType, error) {
```

That is - you don't need to return a slice of results because you are querying by ID which guarantees that there is either exactly one result, or not.

A slice could be if it was `QueryOrgVdcStorageProfile` which doesn't guarantee one object.

Reply:
Can i then rename this method to `GetAdminOrgVdcStorageProfileByID`?

Reply:
Or implement `QueryOrgVdcStorageProfile` with params and `GetAdminOrgVdcStorageProfileByID` that use previous?

Reply:
`GetX` in this SDK usually mean that it is not an object from query endpoint, but from its legitimate endpoint (e.g. `AdminOrg.GetVDCById` would not return a `Query` result, but the VDC struct itself. Query and object endpoints usually return different data that is why we try to keep `Query` and `Get` (to differentiate these things). This is not 100% due to historic reasons, but that is a good thing to follow.

Or - if you want to implement a function that returns all objects then it could be `QueryAdminOrgVdcStorageProfile`. But if you simply leave the code as it is (and not apply ID filter) you will hit a problem - by default it returns only the first page of data (which is usually up to 25 items). It is not easy to spot this on testing envs where not many resources exist, but we have already hit that and fixed some of the problems here. The trick here is to use `client.cumulativeQuery` internally which handles pagination and makes sure all pages are retrieved. (You could look through functions like `QueryEdgeGatewayList` or `QueryAllVdcs` to see how it is implemented.)

Reply:
Agree with you and changed signature.

About `QueryAdminOrgVdcStorageProfile` - i think we need separate issue for discuss.